mava/evaluator.py

```diff
@@ -104,41 +104,68 @@ def not_done(carry: Tuple) -> bool:
 
         return eval_metrics
 
-    def evaluator_fn(trained_params: FrozenDict, key: chex.PRNGKey) -> ExperimentOutput[EvalState]:
-        """Evaluator function."""
+    def parallel_evaluator(
+        carry: Tuple[FrozenDict, chex.PRNGKey], _: Any
+    ) -> Tuple[Tuple[FrozenDict, chex.PRNGKey], ExperimentOutput[EvalState]]:
+        """Evaluates the agent on multiple environments in parallel."""
+        trained_params, key = carry
 
-        # Initialise environment states and timesteps.
         n_devices = len(jax.devices())
+        episodes_per_device = config.arch.num_eval_episodes * eval_multiplier // n_devices
+        parallel_eval_batch_size = min(config.arch.num_envs, episodes_per_device)
 
-        eval_batch = (config.arch.num_eval_episodes // n_devices) * eval_multiplier
-
-        key, *env_keys = jax.random.split(key, eval_batch + 1)
+        # Initialize eval state for the parallel environments.
+        key, *env_keys = jax.random.split(key, parallel_eval_batch_size + 1)
         env_states, timesteps = jax.vmap(env.reset)(jnp.stack(env_keys))
-        # Split keys for each core.
-        key, *step_keys = jax.random.split(key, eval_batch + 1)
-        # Add dimension to pmap over.
-        step_keys = jnp.stack(step_keys).reshape(eval_batch, -1)
+
+        # Split keys for parallel execution in vmap.
+        key, *step_keys = jax.random.split(key, parallel_eval_batch_size + 1)
+        step_keys = jnp.stack(step_keys).reshape(parallel_eval_batch_size, -1)
 
         eval_state = EvalState(
             key=step_keys,
             env_state=env_states,
             timestep=timesteps,
-            step_count=jnp.zeros((eval_batch, 1)),
+            step_count=jnp.zeros((parallel_eval_batch_size, 1)),
             episode_return=jnp.zeros_like(timesteps.reward),
         )
-
+        # Evaluate over a batch of environments in parallel using vmap.
         eval_metrics = jax.vmap(
             eval_one_episode,
             in_axes=(None, 0),
             axis_name="eval_batch",
         )(trained_params, eval_state)
 
-        return ExperimentOutput(
+        return (trained_params, key), ExperimentOutput(
             learner_state=eval_state,
             episode_metrics=eval_metrics,
             train_metrics={},
         )
 
+    def evaluator_fn(trained_params: FrozenDict, key: chex.PRNGKey) -> ExperimentOutput[EvalState]:
+        """Evaluator function"""
+        n_devices = len(jax.devices())
+
+        # Calculate episodes to evaluate per device.
+        episodes_per_device = config.arch.num_eval_episodes * eval_multiplier // n_devices
+        # Determine parallel evaluation batch size.
+        # Limited by architecture's max environments or episodes per device.
+        parallel_eval_batch_size = min(config.arch.num_envs, episodes_per_device)
```

Comment:
Parallel can be a bit confusing, I thought at first this was the `pmap` dim. Please also change below
```suggestion
        # Determine how many eval episodes to vmap over.
        # Limited by architecture's max environments or episodes per device.
        num_vmapped_episodes = min(config.arch.num_envs, episodes_per_device)
```